File: Java/프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.java
class Solution {
    public int solution(int[] arrayA, int[] arrayB) {
        int answer = 0;
        int gcdA = arrayA[0];
        int gcdB = arrayB[0];

        for (int i = 1; i < arrayA.length; i++) {
            gcdA = gcd(gcdA, arrayA[i]);
            gcdB = gcd(gcdB, arrayB[i]);
        }

        if (!isDivisible(arrayA, gcdB)) {
            answer = Math.max(answer, gcdB);
        }

        if (!isDivisible(arrayB, gcdA)) {
            answer = Math.max(answer, gcdA);
        }

        return answer;
    }

    int gcd(int a, int b) {
        return a % b == 0 ? b : gcd(b, a % b);
    }

    boolean isDivisible(int[] arr, int n) {
        for (int i : arr) {
            if (i % n == 0) {
                return true;
            }
        }
        return false;
    }
}

File: Python/프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.py
from math import gcd


def solution(arrayA, arrayB):
    answer = 0

    def isDivisible(arr, num):
        for i in arr:
            if i % num == 0:
                return True
        return False

    a = arrayA[0]
    b = arrayB[0]

    for i in range(len(arrayA)):
        a = gcd(a, arrayA[i])
        b = gcd(b, arrayB[i])

    if not isDivisible(arrayA, b):
        answer = max(answer, b)

    if not isDivisible(arrayB, a):
        answer = max(answer, a)

    return answer